idea/gearexe/frontend/src/app/store/all-activity.ts

```diff
@@ -0,0 +1,38 @@
+import { atom } from 'jotai';
+
+const RouterEvents = {
+  programCreated: 'ProgramCreated',
+  codeGotValidated: 'CodeGotValidated',
+  codeValidationRequested: 'CodeValidationRequested',
+  blockCommitted: 'BlockCommitted',
+} as const;
+
+type RouterEvent =
+  | { type: typeof RouterEvents.programCreated; actorId: string; codeId: string }
+  | { type: typeof RouterEvents.codeGotValidated; codeId: string; valid: boolean }
+  | { type: typeof RouterEvents.codeValidationRequested; codeId: string; blobTxHash: string }
+  | { type: typeof RouterEvents.blockCommitted; hash: string };
+
+const WrappedVaraEvents = {
+  approval: 'Approval',
+  transfer: 'Transfer',
+} as const;
+
+type WrappedVaraEvent =
+  | { type: typeof WrappedVaraEvents.approval; owner: string; spender: string; value: string }
+  | { type: typeof WrappedVaraEvents.transfer; from: string; to: string; value: string };
+
+type AllEvents = RouterEvent | WrappedVaraEvent;
+
+type EventsBlock = {
+  blockHash: string;
+  blockNumber: number;
+  timestamp: number;
+  events: AllEvents[];
+};
+
+type AllActivity = EventsBlock[];
+
+const allActivityAtom = atom<AllActivity>([]);
+
+export { allActivityAtom, RouterEvents, WrappedVaraEvents, type AllEvents, type EventsBlock };
```

Comment:
To have a separate `store` folder seems too complex to me. `Activity` is a feature by itself, therefore all related code should be collocated in features folder.

Also, I believe it would be a good decision to prettify the public api of global state that we expose. The interface that we're communicating with should be pretty much `useActivity` and `useSetActivity`, specific `jotai` realization should be strictly encapsulated.